app/src/main/java/org/dhis2/data/location/LocationProviderImpl.kt

```diff
@@ -46,7 +44,7 @@ class LocationProviderImpl(val context: Context) : LocationProvider {
             onLocationDisabled()
             requestLocationUpdates(onNewLocation)
         } else {
-            locationManager.getLastKnownLocation(locationProvider).apply {
+            locationManager.getLastKnownLocation(locationProvider!!).apply {
```

Comment:
can not be null?

Reply:
Nop, the if else is already checking if it is not null